### ISSUE TITLE
apt install casper-client installs 1.4.1 already which seems to be incompatible with 1.3.4

### ISSUE THREAD

**Issue by GuybrushX** [make-software/how-to-casperlabs#51]
I installed a brand new mainnet validator including casper-client. It seems that the new apt package isn't compatible with the current version because it's newer than the casper node.

casper-client installed: 1.4.1
casper node installed: 1.3.4

Issuing the following command returns an error:
casper-client query-global-state --node-address http://127.0.0.1:7777 -k <PublicKey> -s <StateRootHash>
{
  "code": -32601,
  "message": "Method not found"
}

The docs should mention to take care of the casper-client, too if an upgrade is staged but only after the node was upgraded.

Maybe make casper-client part of the upgrade process to always be in sync with the casper node? Otherwise casper-client needs to be installed after the upgrade was applied or it will be out of sync.

Or block the release of new casper-client versions until the network is upgraded but since the testnet and the mainnet will always be out of sync for some time that is also not the best option.

**Comment by mssteuer** [comment]
Assigning to @mrkara .

These write-ups are specifically about installing the `casper-node`, I'm not sure that including detailed instructions for a client such as `casper-client` belong there, or potentially make things more confusing. Maybe just a notice that says "Make sure that your `casper-client` version matches the version of the `casper-node` you're trying to interact with" ?

**Comment by GuybrushX** [comment]
TL;DR:
1. It was an edge case
2. Update the old-style commands to the new-style
3. Maybe add the error message from my initial post to the Validator FAQ and link to the FAQ for troubleshooting?

FYI: The solution was to downgrade `casper-client` to an older version.

AFAIR Joe Sacher explained that this was a bug and normally shouldn't happen. IMHO a generic note probably doesn't help because how should someone know that the versions are compatible or not? Even right now the following versions are installed as the latest versions:
`casper-node: 1.4.4`
`casper-client: 1.4.3`

-> so they don't match and it could easily lead to: oh, my `casper-client` is outdated - but it's the latest version.

Since the guides are also installing and using `casper-client` I think it should also make sure that the commands are working.

The issue in this specific case was that 'query-state' was renamed to 'query-global-state' IIRC and that the newer `casper-client` with the renamed subcommand was installed already before the `casper-node` was rolled out.

I was just checking and I think this incompatibility blocked the section "Bond to the network" for example because the commands are still using the old sub-command `query-state` instead of the new `query-global-state`.

I'm also not sure what's the best solution here except the following:
In a "perfect" world newer `casper-client` versions should be fully backwards compatible to avoid such issues.

**Comment by GuybrushX** [comment]
Alright, than I'm closing it here. Feel free to forward it to the other team if you think it makes sense.
